### PR TITLE
Updates for SFTPPlus version 4.0.0 and newer.

### PR DESCRIPTION
**Pull request by dumol**
Updated `README` and `Dockerfile` to reference newer supported OS'es and SFTPPlus 4.0.0.

Log to stdout only and start in foreground. All logs available to `docker logs` while `log/` sub-dir should be empty.

Generate the SSL self-signed cert and the SSH keys at Docker build time without initializing configuration.

Updated the included `configuration/server.ini` with latest additions to an automatically initialized default config.

To check, please build an image and run a container following the instructions.

### CODE REVIEW

File: sftpplus-docker-setup.sh [hunk 1]
```diff
@@ -31,7 +33,7 @@ ln -s sftpplus-${SFTPPLUS_OS}-x64-${SFTPPLUS_VERSION} sftpplus
 # Initialize SFTPPlus configuration.
 # This will generate the SSH keys and the self signed certificates.
 cd sftpplus
-./bin/admin-commands.sh initialize
+./bin/admin-commands.sh initialize --init-admin admin --init-password pass
```
Comment (adiroiban) [inline sftpplus-docker-setup.sh]: I don't know if for docker usage, customers will want to initialized SFTPPlus for each image.

The old initialize command was there only to create a new self signed cert and ssh keys. 

I think that we should keep the server.ini file, and instead use explicit command to create the SSH keys like

```
./bin/admin-commands.sh generate-ssh-key --key-skip --key-type rsa --key-size 2048 --key-file /path/to/rsa_host
```

now we don't have a command to generate a self-signed key.

I will introduce it in v4.3.0 and we can use it here.

**Review by adiroiban — CHANGES_REQUESTED** [review]
for docker usage, I think that customers will want to have direct control to the server.ini and not use the web manage page for configuration

I will add support for generating a self-signed certificate from the command line and then we can resume working on this PR.

Thanks!

File: sftpplus-docker-setup.sh (hunk 1 above)
Comment (adiroiban) [reply sftpplus-docker-setup.sh]: one option is to generate the self-signed certificate using the openssl command :)

Comment (dumol) [reply sftpplus-docker-setup.sh]: Well, this repo is for evaluation purposes, so I'm fine with how things stand.

Of course, the above would be a nice addition, to be documented here as well. But I would still keep this repo self-sufficient, not requiring extra stuff to quickly fire up a container to try.

Comment (dumol) [reply sftpplus-docker-setup.sh]: Came back to this now that 4.3.0 is released, but I can't find any `server.ini` file in the trial packages (I've tried the `rhel8` one). What do you mean by "we should keep the server.ini file"? Thanks

**Comment by adiroiban** [comment]
needs-change

I think with docker, we can disable logging into a file, and just enable standard output logger.

Also, we should have `CMD [ "bin/admin-commands.sh", "start-in-foreground" ]`

**Comment by dumol** [comment]
> I think with docker, we can disable logging into a file, and just enable standard output logger.
> 
> Also, we should have `CMD [ "bin/admin-commands.sh", "start-in-foreground" ]`

Done the above.

Also, self-signed SSL cert and SSH keys are generated without reinitializing configuration.

Have updated `server.ini` for the above + new additions that I've picked up from newly-initialized configs. Also, I think there's no sense in disabling `root` logins as long as OS accounts are disabled, so I have yanked those sections too. Please double-check!

**needs-review**

File: configuration/server.ini [hunk 2]
```diff
@@ -63,15 +64,24 @@ description = Deny all administrator accounts.
...
+[authentications/ban-ip-uuid]
 enabled = Yes
+type = ip-time-ban
+name = Ban IP with multiple failures
+description = Will ban the source IP for a certain time on multiple failures.
+ban_interval = 600
+ban_after_count = 10
+
+
+[databases/default-sqlite-db-uuid]
```
Comment (adiroiban) [inline configuration/server.ini]: This section can be removed as [databases/] are no longer used in v4.

Or maybe replace it with `[resources/DEFAULT-SQLITE]`

I will commit an update to the config

**Review by adiroiban — APPROVED** [review]
Great work. Thanks.

I have pushed a few changes to the default config file.

All good to merge and release on docker-hub based on 4.4.0

**Comment by dumol** [comment]
> 
> I have pushed a few changes to the default config file.

Thanks! I have added some more changes, but mostly cosmetical. The only one of notice would be disabling TLS 1.0 and 1.1 for LM in the included config.

> 
> All good to merge and release on docker-hub based on 4.4.0

4.0.0 - 4.4.0 trial containers were generated from this branch. Not with latest changes in regards to logging though…

**Pull request merged**